### PR TITLE
ddl: save and restore auto_random id when rename table (#18212)

### LINKED ISSUE #18211: rename auto_random table lead to duplicate entry
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

```sql
create table t (a bigint auto_random primary key);
insert into t values ();
...  -- repeat 100 times
insert into t values ();
alter table t rename to newdb.t;
alter table newdb.t rename to t;
insert into t values ();
...  -- repeat 100 times BANG!
insert into t values ();
```

### 2. What did you expect to see? (Required)

Query Ok.

### 3. What did you see instead (Required)

```console
Error: Duplicate entry 'xxxxx' for key 'PRIMARY'
```

### 4. Affected version (Required)

master, v3.1.0, v4.0.0

### 5. Root Cause Analysis

`auto_random` ID should be kept when renaming tables.

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #18212 to release-4.0

---

### What problem does this PR solve?

Issue Number: close #18211 

Problem Summary: 

`auto_random` ID should be kept when renaming tables.

### What is changed and how it works?

What's Changed:

- Save and restore `auto_random` ID in `onRenameTable()`.
- Add a test for renaming `auto_random` table

How it Works:

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 

- fix duplicate entry when renaming `auto_random` tables.

### CODE REVIEW

File: executor/ddl_test.go [hunk 1]
```diff
@@ -872,13 +877,77 @@ func (s *testAutoRandomSuite) TestAutoRandomBitsData(c *C) {
...
 
+<<<<<<< HEAD
 	tk.MustExec("create table t (a tinyint primary key auto_random(2), b int)")
 	tk.MustExec("insert into t values (1, 2)")
 	tk.MustExec("update t set a = 31 where a = 1")
 	_, err = tk.Exec("insert into t (b) values (0)")
 	c.Assert(err, NotNil)
 	c.Assert(err.Error(), Equals, autoid.ErrAutoRandReadFailed.GenWithStackByArgs().Error())
 	tk.MustExec("drop table t")
+=======
```
Comment (zimulala) [inline executor/ddl_test.go:888]: @tangenta 
Please fix the conflict.

**Comment by tangenta** [comment]
/run-all-tests
```
[2020-06-29T02:24:24.879Z] ----------------------------------------------------------------------
[2020-06-29T02:24:24.880Z] FAIL: builtin_miscellaneous_vec_test.go:129: testEvaluatorSuite.TestSleepVectorized
[2020-06-29T02:24:24.880Z] 
[2020-06-29T02:24:24.880Z] builtin_miscellaneous_vec_test.go:213:
[2020-06-29T02:24:24.880Z]     c.Assert(result.GetInt64(1), Equals, int64(1))
[2020-06-29T02:24:24.880Z] ... obtained int64 = 0
[2020-06-29T02:24:24.880Z] ... expected int64 = 1
[2020-06-29T02:24:24.880Z] 
[2020-06-29T02:24:24.880Z] 
[2020-06-29T02:24:24.880Z] ----------------------------------------------------------------------
```

**Pull request merged**